src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -322,6 +323,7 @@ public bool IsSolutionDPLEnabled
                 {
                     await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
+                    EnsureCacheInitialized();
```

Comment:
I wanted to keep it consistent with existing behavior, and more so initialization will anyway be done at similar time frame. Also without it, we'll be partially initializing here and then at some point, again call for full initialization, which will not be good idea. What do you think?